tripper/dataset/dataset.py

```diff
@@ -0,0 +1,736 @@
+        d[key] = value
+    else:
+        klst = d[key] if isinstance(d[key], list) else [d[key]]
+        vlst = value if isinstance(value, list) else [value]
+        v = list(set(klst).union(vlst))
+        d[key] = v[0] if len(v) == 1 else sorted(v)
+
+
+def get(
+    d: dict, key: str, default: "Any" = None, aslist: bool = True
+) -> "Any":
+    """Like `d.get(key, default)` but returns the value as a list if
+    `aslist` is True and value is not already a list.
+
+    An empty list is returned in the special case that `key` is not in
+    `d` and `default` is None.
+
+    """
+    value = d.get(key, default)
+    if aslist:
+        return (
+            value
+            if isinstance(value, list)
+            else [] if value is None else [value]
+        )
+    return value
+
+
+def expand_iri(iri: str, prefixes: dict) -> str:
+    """Return the full IRI if `iri` is prefixed.  Otherwise `iri` is
+    returned."""
+    match = re.match(MATCH_PREFIXED_IRI, iri)
+    if match:
+        prefix, name, _ = match.groups()
+        if prefix in prefixes:
+            return f"{prefixes[prefix]}{name}"
+        warnings.warn(f'Undefined prefix "{prefix}" in IRI: {iri}')
+    return iri
+
+
+def read_datadoc(filename: "Union[str, Path]") -> dict:
+    """Read YAML data documentation and return it as a dict."""
+    with open(filename, "r", encoding="utf-8") as f:
+        d = yaml.safe_load(f)
+    return prepare_datadoc(d)
+
+
+def save_datadoc(
+    ts: Triplestore, file_or_dict: "Union[str, Path, dict]"
+) -> dict:
+    """Populate triplestore with data documentation.
+
+    Arguments:
+        ts: Triplestore to load dataset from.
+        file_or_dict: Data documentation dict or name of a YAML file to read
+            the data documentation from.
+
+    Returns:
+        Dict-representation of the loaded dataset.
+    """
+    if isinstance(file_or_dict, dict):
+        d = prepare_datadoc(file_or_dict)
+    else:
+        d = read_datadoc(file_or_dict)
+
+    # Bind prefixes
+    prefixes = get_prefixes()
+    prefixes.update(d.get("prefixes", {}))
+    for prefix, ns in prefixes.items():  # type: ignore
+        ts.bind(prefix, ns)
+
+    # Maps datadoc_labels to type
+    types = {v["datadoc_label"]: k for k, v in dicttypes.items()}
+
+    # Write json-ld data to triplestore (using temporary rdflib triplestore)
+    for spec in dicttypes.values():
+        label = spec["datadoc_label"]
+        for dct in get(d, label):
+            dct = prepare(types[label], dct, prefixes=prefixes)
+            f = io.StringIO(json.dumps(dct))
+            with Triplestore(backend="rdflib") as ts2:
+                ts2.parse(f, format="json-ld")
+                ts.add_triples(ts2.triples())
+
+    # Add statements and datamodels to triplestore
+    save_extra_content(ts, d)
+    return d
+
+
+def prepare_datadoc(datadoc: dict) -> dict:
+    """Return an updated version of dict `datadoc` that is prepared for
+    serialisation to RDF."""
```

Comment:
Prepare with what? Please specify so the documentation is complete for a new user.

Reply:
Updated the documentation. The function is mostly ment for internal use. Only the function listed at the top of the module are actually intended for users.